matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -166,6 +166,8 @@ public MatchResult doWork(List<GPXEntry> gpxList) {
         // Compute all candidates first.
         // TODO: Generate candidates on-the-fly within computeViterbiSequence() if this does not
         // degrade performance.
+        // @kodonnell: it's not currently possible as that'd mean calling queryGraph.lookup
```

Comment:
@stefanholder I probably do not understand this yet deep enough, but why do you think this would mean calling queryGraph.lookup multiple times?

Furthermore: creating a QueryGraph is cheap so we could call it then multiple times but the resulting QueryResults are independent.